### PR TITLE
Duotao/registration frontend

### PR DESCRIPTION
**Pull request by taoduo**
## Operations Completed
1. Registration cases: good, exists verified, exists not verified, invalid info, server error
2. Login cases: good, wrong email / password, not verified, server error
3. Password Reset cases: good, username not exists, username empty, server error
4. Resend Verification cases: good, server error 
(I also included 400 and 404 for this but they should not happen on the frontend UI)

Are these cases valid? Did I miss any?

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ exports.setup = function (app) {
         .post(users.registerUser,
               mailer.sendRegistrationEmail,
               responder.formatCurrentUser);
-
+    app.route('/api/resendRegistration').post();
```
Comment (dpikt) [inline app/routes.js]: Is this used?

File: public/scripts/main.js [hunk 2]
```diff
@@ -256,23 +256,99 @@ hitsTheBooks.controller('accountAccessController', function($scope, $rootScope,
 
   // Login
   $scope.loginData = { username: '', password: '' };
-
+  $scope.signinAlert = 0;
```
Comment (dpikt) [inline public/scripts/main.js]: I would suggest using an enum for the value of signinAlert, just to make it clearer what each case means

File: app/routes.js (hunk 1 above)
Comment (taoduo) [inline app/routes.js]: No. I did not know I added it. I will remove it.

File: public/scripts/main.js (hunk 2 above)
Comment (taoduo) [inline public/scripts/main.js]: All right.

**Comment by dpikt** [comment]
Awesome! LGTM

**Pull request merged**